server/src/controllers/auth.controller.js

```diff
@@ -39,9 +39,46 @@ const signup = ( req, res) => {
         });
     });
 };
+const signin = (req,res) => {
+    User.findOne({ email: req.body.email})
+    .exec ( (error, user) => {
+        if(error) return res.status(400).json({ error});
+        if( user){
+
+            if(user.authenticate(req.body.password)){
+                const token = jwt.sign({ _id: user._id, role: user.role }, process.env.JWT_SECRET, { expiresIn: '1h'});
+                const { _id, firstName, lastName, email, role, fullName } = user;
+                res.status(200).json({
+                    token,
+                    user: {
+                        _id, firstName, lastName, email, role, fullName
+                    }
+                });
+            
+            } else {
+                return res.status(400).json({
+                    message: 'Invalid Password'
+                })
+            }
+
+        } else {
+            return res.status(400).json({message: 'Something went wrong'});
+        }
+    })
+}
+
+const requireSignin = ( req, res, next) => {
+    const token = req.headers.authorization.split(" ")[1];
+    const user = jwt.verify(token, process.env.JWT_SECRET);
+    req.user = user;
+    next();
+}
+
```

Comment:
This seems like a middleware more, so keep it in it's desired location which means the middleware directory, as a separate file